code/modules/client/preferences.dm

```diff
@@ -303,7 +303,7 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 				return FALSE
 
 			color_list[requested_preference_index] = new_color
-			if (!update_preference(requested_preference, color_list))
+			if (!update_preference(requested_preference, color_list.Join(";")))
```

Comment:
If you're gonna store these as a string anyways, why not use BYOND's params procs?